### PR TITLE
pulley: Implement SIMD `splat` instruction

### PR DESCRIPTION
**Pull request by alexcrichton**
Gets a few spec tests and CLIF tests passing

cc #9783

### CODE REVIEW

File: cranelift/codegen/src/isa/pulley_shared/lower.isle [hunk 1]
```diff
@@ -771,3 +771,12 @@
 
 (rule 0 (lower (has_type (fits_in_32 _) (iabs a))) (pulley_xabs32 (sext32 a)))
 (rule 1 (lower (has_type $I64 (iabs a))) (pulley_xabs64 a))
+
+;;;; Rules for `split` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
```
Comment (abrown) [inline cranelift/codegen/src/isa/pulley_shared/lower.isle:775]:
```suggestion
;;;; Rules for `splat` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
```

File: pulley/src/interp.rs [hunk 2]
```diff
@@ -2695,6 +2695,42 @@ impl OpVisitor for Interpreter<'_> {
         self.state[dst].set_u128(val);
         ControlFlow::Continue(())
     }
+
+    fn vsplatx8(&mut self, dst: VReg, src: XReg) -> ControlFlow<Done> {
+        let val = self.state[src].get_u32() as u8;
```
Comment (abrown) [inline pulley/src/interp.rs:2700]: We may need `get_u8` and `get_u16` eventually?

Comment (abrown) [reply pulley/src/interp.rs:2700]: ...just mulling over whether discarding the upper bits here is going to be problematic &mdash; probably not?

**Review by abrown — APPROVED** [review]
LGTM!

File: pulley/src/interp.rs (hunk 2 above)
Comment (alexcrichton) [reply pulley/src/interp.rs:2700]: I've been wondering the same yeah, I'll probably add these in a follow-up

**Pull request merged**